lib/elixir/lib/calendar/time.ex

```diff
@@ -522,6 +525,18 @@ defmodule Time do
     {:ok, time}
   end
 
+  def convert(
+        %{
+          hour: _,
+          minute: _,
+          second: _,
+          microsecond: _
+        },
+        _calendar
+      ) do
+    {:error, :unrecognized_time_format}
```

Comment:
What would be a proper reason for this error?

Reply:
I don't think we should have this clause, if the 1st argument isn't a valid time map the function should crash.

Reply:
I think the idea is the `convert!/2` crashes if that's the case, but as of now, `convert!/2` never raises because convert/2 never returns `{:error, _}`

Reply:
Sure. But the problem is in this clause we are basically handling a situation that someone has a map with incorrect shape (incorrect microseconds) and instead of gracefully handling that, we should crash. If you wanna look at it from the types angle, we're handling a situation that shouldn't be possible given the 1st argument should be of `Calendar.time()` type.